torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -98,6 +98,19 @@ class ProcessGroupNCCL : public ProcessGroup {
   };
 
   // Constructor will also check the number of available GPUs in the system
+  //
+  // Group support:
+  //
+  // In order to support multiple NCCL process groups, each of which has
+  // different group ranks, we need to use groupName to identify each group
+  // to ensure the correct behavior. In other words, each process group that
+  // has different group ranks needs to have a different and unique groupName
+  // to avoid clashing into undefined behaviors.
```

Comment:
This doesn't mention ordering. The group name is used to ensure that multiple NCCL process groups created either concurrently, or otherwise out of order, can still find each other. Then when the same group name is used, they are required to be created in the same global order.